utils/run_test.sh

```diff
@@ -307,6 +307,7 @@ if [ -d "/mnt/daos" ]; then
 "
         done
         produce_output 1 "$fail_msg"
+        # This IF test should probably not be here, as it breaks pipeline-lib
```

Comment:
I believe this is here because outside of CI, users/devs want test failures to result in a `!0` exit code.  Perhaps for non-junit parsing scripting, etc.  But in CI, we don't want just test failures to result in a `!0` exit code because we interpret a test-running-framework exiting `!0` as the framework failing, not simply some tests didn't pass.  We have JUnit files to be able to determine test failures from.

Reply:
The behavior of the pipeline-lib is currently to fail the stage on a non-zero return code.
To make it mark the stage as unstable on a non-zero return code would be a minor change to the PR in progress.
That would be global though to all code that uses run-test.

With the pipeline-lib PR there is a fix that if no Junit result files are found or any expected valgrind files are not present the stage will be failed.  With pipeline-lib master that error condition is ignored.

We can have pipeline-lib determine from the test results retrieved to mark the test as failed if it detects something from the test results.  With pipeline-lib master, for Unit tests we are reporting the test results with out valgrind results to Junit and allowing Junit to mark the stage as unstable, and reporting a success to github for the stage.

Reply:
To totally fix the color labeling for the unstable V.S. failure issue would take a bigger change than is in the pipeline-lib PR currently.

We would need to move the retrieval of he test results and do the pre-check of them from unitTestPost,.groovy to UnitTest.groovy, and make a corresponding change to for functional testing.

To totally control the stage color of Unstable, all result pre-checks before reporting to github need to be in the steps section.

Reply:
Removing the if around the exit one is needed, or I can not get the pipeline-lib methods to work as intended for the memcheck stage.
The impact of removing that if is that until the ftest.sh is fixed to only report a failure on test setup errors, that we will get some stages set to FAILURE for test failures when you want them to be set to UNSTABLE, and that can be an action item for the test maintainers to fix.

To fix the rest of pipeline-lib to properly set the status to UNSTABLE for test failures for Unit and functional tests and obey the ignore_failure=true, will take more work on pipeline-lib. which I will start testing with a fork of the pipeline-linb PR.